pycardano/wallet.py

```
@@ -0,0 +1,1938 @@
+        )
+
+        return utxo_creator
+
+    else:
+        logger.warn(
+            "Fetching UTxO creators (sender) is only possible with Blockfrost Chain Context."
+        )
+
+
+def get_utxo_block_time(utxo: UTxO, context: ChainContext) -> int:
+    """Get the block time at which a UTxO was created.
+
+    Args:
+        utxo (UTxO): The UTxO to get the block time of.
+        context (ChainContext): The context to use for the query. For now must be BlockFrost.
+
+    Returns:
+        int: The block time at which the UTxO was created.
+
+    """
+    if isinstance(context, BlockFrostChainContext):
+        block_time = context.api.transaction(str(utxo.input.transaction_id)).block_time
+
+        return block_time
+
+    else:
+        logger.warn(
+            "Fetching UTxO block time is only possible with Blockfrost Chain Context."
+        )
+
+
+def get_stake_info(stake_address: Union[str, Address], context: ChainContext) -> dict:
+    """Get the stake info of a stake address from Blockfrost.
+    For more info see: https://docs.blockfrost.io/#tag/Cardano-Accounts/paths/~1accounts~1{stake_address}/get
+
+    Args:
+        stake_address (Union[str, Address]): The stake address to get the stake info of.
+        context (ChainContext): The context to use for the query. For now must be BlockFrost.
+
+    Returns:
+        dict: Info regarding the given stake address.
+    """
+    if isinstance(context, BlockFrostChainContext):
+
+        if isinstance(stake_address, str):
+            stake_address = Address.from_primitive(stake_address)
+
+        if not stake_address.staking_part:
+            raise TypeError(f"Address {stake_address} has no staking part.")
+
+        try:
+            return context.api.accounts(str(stake_address)).to_dict()
+        except ApiError:
+            return {}
+
+    else:
+        logger.warn(
+            "Retrieving stake address information is only possible with Blockfrost Chain Context."
+        )
```

Can we raise an exception when the wrong context is passed? Raising an explicit exception is probably safer than returning a None value. What do you think?

---

Agreed. There are a few places where similar logic also might apply:

`Wallet.stake_info` and other stake-related methods warn when stake address is not yet registered. Not sure if it makes more sense to leave it as a warning and return nothing or raise and error of some sort.

`Wallet.sync()` warns when an address has not transacted on the chain yet. I think leaving it as a warning should be fine.
``

---

Makes sense, for these two cases we can use warning. Thank you!